src/app/components/CoinStatistics/CoinStatistics.tsx

```diff
@@ -22,14 +31,42 @@ ChartJS.register(
   Legend
 );
 
-const CoinStatistics = () => {
- 
+const CoinBlock = ({ data }) => {
+  const coinName = data.name;
+  const symbol = data.symbol;
+  const currentPrice = Number(data.current_price).toFixed(2);
+  const oneDayChange = Number(data.price_change_percentage_24h).toFixed(2);
+  const coinImage = data.image;
+
   return (
-    <MainWrapper>
-      <CoinCarousel>This is where the coin selection will go</CoinCarousel>
-      <MainGraphWrapper>
+    <CoinWrapper>
+      <LeftSection>
+        <img className="h-10" src={coinImage} alt="" />
+      </LeftSection>
+      <MiddleSection>
+        <div>
+          {coinName} ({symbol})
+        </div>
+        <div>{currentPrice}</div>
+      </MiddleSection>
+      <RightSection>{oneDayChange}</RightSection>
+    </CoinWrapper>
+  );
+};
 
-      </MainGraphWrapper>
+CoinBlock.propTypes = {
+  data: PropTypes.object,
+};
+
+const CoinStatistics = ({ data }) => {
+  return (
+    <MainWrapper>
+      <CoinCarousel>
+        {data.map((coin) => (
+          <CoinBlock key={uid()} data={coin} />
```

Comment:
this uid is changing on each render, put this, at the root source of the data so it doesn;t change

Reply:
I don't see where else I should leave it right now.

On page I fetch the data and then pass it to CoinStatistics and CoinDetails. CoinDetails maps immediatly since I currently won't be using it for anything else. CoinStatistics gets the entire array because I'm going to use it for graph info.

What is different between the CoinStatistics component and CoinDetails on app/page?